cpp/include/raft/sparse/linalg/spmm.hpp

```diff
@@ -60,8 +60,12 @@ void spmm(raft::resources const& handle,
 {
   bool is_row_major = detail::is_row_major(y, z);
 
-  auto z_tmp_view = raft::make_device_strided_matrix_view<ValueType, IndexType, LayoutPolicyZ>(
-    z.data_handle(), z.extent(0), z.extent(1), is_row_major ? z.stride(0) : z.stride(1));
+  auto z_tmp_view =
+    is_row_major
+      ? raft::make_device_strided_matrix_view<ValueType, IndexType, layout_c_contiguous>(
+          z.data_handle(), z.extent(0), z.extent(1), is_row_major ? z.stride(0) : z.stride(1))
```

Comment:
We already know at that point if the layout is row major:
```suggestion
          z.data_handle(), z.extent(0), z.extent(1), z.stride(0))
```